pyteal/ast/subroutine_test.py

```diff
@@ -110,6 +126,61 @@ def fnWithMixedAnns3(a: Expr, b: ScratchVar):
         ("all sv's 1", fnWithSVAnnotations, [sv, sv], None),
         ("all sv's but strings", fnWithSVAnnotations, [s, s], TealInputError),
         ("all sv's but ints", fnWithSVAnnotations, [x, x], TealInputError),
+        (
+            "all abi's 1",
+            fnWithABIAnnotations,
+            [av_byte, av_u32_static_arr, av_bool_dym_arr],
+            None,
+        ),
+        (
+            "all abi's but ints 1",
+            fnWithABIAnnotations,
+            [x, av_u32_static_arr, av_bool_dym_arr],
+            TealInputError,
+        ),
+        (
+            "all abi's but ints 2",
+            fnWithABIAnnotations,
+            [x, av_u32_static_arr, x],
+            TealInputError,
+        ),
+        ("all abi's but ints 3", fnWithABIAnnotations, [x, x, x], TealInputError),
+        (
+            "all abi's but sv's 1",
+            fnWithABIAnnotations,
+            [sv, av_u32_static_arr, av_bool_dym_arr],
+            TealInputError,
+        ),
+        (
+            "all abi's but sv's 2",
+            fnWithABIAnnotations,
+            [av_byte, av_u32_static_arr, sv],
+            TealInputError,
+        ),
+        (
+            "all abi's but sv's 3",
+            fnWithABIAnnotations,
+            [av_byte, sv, av_u32_static_arr],
+            TealInputError,
+        ),
+        (
+            "all abi's but wrong typed 1",
+            fnWithABIAnnotations,
+            [av_u32_static_arr, av_u32_static_arr, av_bool_dym_arr],
+            TealInputError,
+        ),
+        (
+            "all abi's but wrong typed 2",
+            fnWithABIAnnotations,
+            [av_u32_static_arr, av_u32_static_arr, av_bool_dym_arr],
+            TealInputError,
+        ),
+        (
+            "all abi's but wrong typed 3",
+            fnWithABIAnnotations,
+            [av_u32_static_arr, av_u32_static_arr, av_bool_dym_arr],
```

Comment:
As far as I can tell, "all abi's but wrong typed 1", "all abi's but wrong typed 2", and "all abi's but wrong typed 3" are the same inputs?